lib/chef/knife/azurerm_base.rb

```diff
@@ -146,18 +174,18 @@ def check_token_validity(token_details)
 
       def validate_azure_login
         err_string = "Please run XPLAT's 'azure login' command OR specify azure_tenant_id, azure_subscription_id, azure_client_id, azure_client_secret in your knife.rb"
-        if Chef::Platform.windows?
+        if Chef::Platform.windows? && (is_old_xplat? || is_WCM_env_var_set?)
           # cmdkey command is used for accessing windows credential manager
           xplat_creds_cmd = Mixlib::ShellOut.new("cmdkey /list | findstr AzureXplatCli")
           result = xplat_creds_cmd.run_command
           if result.stdout.nil? || result.stdout.empty?
             raise err_string
           end
-        else
-          home_dir = File.expand_path('~')
-          if !File.exists?(home_dir + "/.azure/accessTokens.json") || File.size?(home_dir + '/.azure/accessTokens.json') <= 2
-            raise err_string
-          end
+          return
```

Comment:
Even if this was still an `else` block, we would get here on Windows because while `Chef::Platform.windows?` would be true, but in this case `(is_old_xplat? || is_WCM_env_var_set?)` would not be.

A comment before the `if` that says something like "Older versions of the Azure CLI on Windows stored credentials in a unique way. Newer versions use the same pattern across platforms." would help.